index.js

```diff
@@ -106,6 +113,12 @@ module.exports = postcss.plugin('postcss-css-variables', function(options) {
 				});
 				variableRootRule.append(varDecl);
 
+				// mark JS-injected variables for removal if `cleanInjectedVariables`
```

Comment:
Just above lines 106-114, we insert a root note with the variables. We can probably just add an if-statement `if(!opts.preserveInjectedVariables)` instead of having to remove the decls we just added.

Reply:
I'm waiting until the end to remove injected variable declarations so that they're available in the AST to calculate fallbacks for `var()` declarations as expected.

Reply:
@akdetrick I think the `map` should be sufficient though? We don't look in the AST for variables after finding them all in the beginning